sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanHelper.scala

```diff
@@ -139,15 +138,21 @@ trait AdaptiveSparkPlanHelper {
   }
 
   /**
-   * Returns a cloned [[SparkSession]] with adaptive execution disabled, or the original
-   * [[SparkSession]] if its adaptive execution is already disabled.
+   * Returns a cloned [[SparkSession]] with all specified configurations disabled, or
+   * the original [[SparkSession]] if all configurations are already disabled.
    */
-  def getOrCloneSessionWithAqeOff[T](session: SparkSession): SparkSession = {
-    if (!session.sessionState.conf.adaptiveExecutionEnabled) {
+  def getOrCloneSessionWithConfigsOff(
```

Comment:
Since this method is not only for AQE now, could you move this method into a more suitable place, e.g., `object SparkSessoin` or somewhere?

Reply:
+1 move to other general object.

Reply:
Sounds good it makes sense to me, moved to `object SparkSessoin`.

Reply:
I know this is old, but why all the configurations (e.g., AQE) must be disabled for CacheManager?

Reply:
That's because performance regression can happen. Could you check the previous discussion, e.g., https://github.com/apache/spark/pull/29804#issuecomment-701110070 ?

Reply:
@maropu Sorry for the late. 

Can we make these disable configs more public ? i.e. expose a new SQL config to user, and the exists value as defualt.

Reply:
I feel introducing the configs to enable them (i.e. allow user to enable AQE for cached query, or allow user to enable aoth bucketed scan for cached query) is dangerous, as user can cause correctness bug to their pipeline if using them blindly.

Reply:
@c21 I have such thought at first, but cann't find a negative case. Can you point out a case that can cause correctness bug ? 

If I don's miss something, it just affect the perfermance about extra shuffle. For correctness, let's assuming a cache plan with AQE enabled:
* For lazy cache. the AQE framework will ensure the correctness of the new query with the cached plan .
* For force cache. if the output paritioning or ordering of cached plan has been affected by AQE then Spark will use `EnsureRequirements` to promise the correctness.

Reply:
Is this related to correctness? I thought this was performance related because they can change output partitions implicitly.

Reply:
Could you move this discussion into https://issues.apache.org/jira/browse/SPARK-35332 ? I thinks the jira ticket is related to this topic.